File: ru.kmorozov.api.onedrive/src/main/java/ru/kmorozov/onedrive/client/resources/Authorisation.java
package ru.kmorozov.onedrive.client.resources;

import com.google.api.client.util.Key;

public class Authorisation {

    @Key("token_type")
    private String tokenType;
    @Key("expires_in")
    private int expiresIn;
    @Key("scope")
    private String scope;
    @Key("access_token")
    private String accessToken;
    @Key("refresh_token")
    private String refreshToken;
    @Key("user_id")
    private String userId;
    @Key("error")
    private String error;
    @Key("error_description")
    private String errorDescription;

    public String getTokenType() {
        return this.tokenType;
    }

    public int getExpiresIn() {
        return this.expiresIn;
    }

    public String getScope() {
        return this.scope;
    }

    public String getAccessToken() {
        return this.accessToken;
    }

    public String getRefreshToken() {
        return this.refreshToken;
    }

    public String getUserId() {
        return this.userId;
    }

    public String getError() {
        return this.error;
    }

    public String getErrorDescription() {
        return this.errorDescription;
    }
}

File: ru.kmorozov.api.onedrive/src/main/java/ru/kmorozov/onedrive/client/resources/Authorisation.kt
package ru.kmorozov.onedrive.client.resources

import com.google.api.client.util.Key

class Authorisation {

    @Key("token_type")
    val tokenType: String? = null
    @Key("expires_in")
    val expiresIn: Int = 0
    @Key("scope")
    val scope: String? = null
    @Key("access_token")
    val accessToken: String? = null
    @Key("refresh_token")
    val refreshToken: String? = null
    @Key("user_id")
    val userId: String? = null
    @Key("error")
    val error: String? = null
    @Key("error_description")
    val errorDescription: String? = null
}
